dali/ft/filecopy.cpp

```diff
@@ -1943,9 +1998,11 @@ void FileSprayer::cloneHeaderFooter(unsigned idx, bool isHeader)
     PartitionPoint & next = * new PartitionPoint;
     //NB: headerSize include the size of the xmlHeader; size includes neither header or footers.
     if (isHeader)
+        // Set offset to the XML header
         next.inputOffset = curSrc.headerSize - curSrc.xmlHeaderLength;
     else
-        next.inputOffset = curSrc.headerSize + curSrc.size;
+        //Set offset to the XML footer
+        next.inputOffset = curSrc.size + curSrc.xmlHeaderLength;
```

Comment:
For the first question: It is a really hard to follow code. Somewhere strange things are going. Sometimes the BOM skipped, sometimes not. The original calculation seems good, but caused skewed footers (I traced it), that's why I change it. If you have any non-xml input file candidate or structure in your mind, please attach it to the JIRA and will test/trace it.

For the second question: Yes I tested it with plain ASCII and utf-8 with 3 bytes of BOM. The result is good.  So somewhere any strange magic happen in this abandoned legacy code. 

If you give me a non-xml file and this code break it then I will raise a different JIRA to fix it.